pkg/agent/route/route_linux.go

```diff
@@ -1420,35 +1420,32 @@ func (c *Client) AddLoadBalancer(svcIP net.IP) error {
 		mask = net.IPv6len * 8
 	}
 
-	route := generateRoute(svcIP, mask, gw, linkIndex, netlink.SCOPE_UNIVERSE)
+	route := generateRoute(externalIP, mask, gw, linkIndex, netlink.SCOPE_UNIVERSE)
 	if err := c.netlink.RouteReplace(route); err != nil {
-		return fmt.Errorf("failed to install route for LoadBalancer ingress IP %s: %w", svcIP.String(), err)
+		return fmt.Errorf("failed to install route for external IP %s: %w", externalIPStr, err)
 	}
-	klog.V(4).InfoS("Added LoadBalancer ingress IP route", "route", route)
-	c.serviceRoutes.Store(svcIP.String(), route)
-
+	c.serviceRoutes.Store(externalIP.String(), route)
```

Comment:
```suggestion
	c.serviceRoutes.Store(externalIPStr, route)
```
since you created a string intentionally to avoid duplicate conversion.

Reply:
Thanks for reminding, I missed this.